creator-node/src/snapbackSM.js

```diff
@@ -589,4 +584,45 @@ class SnapbackSM {
   }
 }
 
+/**
+ * Ensure a sync for (syncType, userWallet, secondaryEndpoint) can only be enqueued once
+ * This is used to ensure multiple concurrent sync tasks are not being redundantly used on a single user
+ * Implemented with an in-memory map of string(syncType, userWallet, secondaryEndpoint) -> object(syncJobInfo)
+ *
+ * @dev We maintain this map to maximize query performance; Bull does not provide any api for querying
+ *    jobs by property and would require a linear iteration over the full job list
+ */
+class SyncDeDuplicator {
```

Comment:
I don't think this really has to be a class at all, all of these seem like member functions of Snapback to me but this is a non-blocking NIT

Reply:
yeah i thought about that, just getting tired of all the massive flag classes with all these functions with no hierarchical grouping. i get your point, intent was just to have a more readable and more logical grouping of code